ecommerce/serializers.py

```diff
@@ -367,15 +367,26 @@ def _update_items(cls, basket, items):
             item = items[0]
             product_id = item.get("product_id")
             run_ids = item.get("run_ids")
-            product = models.Product.objects.get(id=product_id)
+            try:
+                # product_id = product_id
+                product_id = int(product_id)
+                product = models.Product.all_objects.get(id=product_id)
```

Comment:
reason for filtering over non live products here?

Reply:
not any specific reason, I was thinking its for authors what if he wants to search about inactive courses.

Reply:
I think, over checkout page, we should filter over live products only. @pdpinch thoughts

Reply:
Currently if you try to checkout with an inactive product, you get an error. We should keep this behavior.

Reply:
However, if you try to checkout with a course run id that is closed for enrollment, we attempt to offer you other runs that are open for enrollment, for example: https://xpro.mit.edu/checkout/?product=1